### PR TITLE
func: add ? operator support for map types

### PR DESCRIPTION
**Pull request by JLDLaughlin**
The `?` operator checks if a map contains a given key, returning true
if the key is found and false otherwise. For example:

        SELECT '{a=>1, b=>2}'::map[text=>int] ? 'a'

Returns true.

As we only support text keys for maps, the expression provided as the
right hand side of the operator must be of type text.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4840)

### CODE REVIEW

File: test/sqllogictest/map.slt [hunk 1]
```diff
@@ -83,3 +84,24 @@ SELECT '{"a"=>"hello there!}'::map[text=>text]
 
 query error nested map types not yet supported
 SELECT '{a=>{b=>c}}'::map[text=>map[text=>text]]
+
+# Test map operators.
```
Comment (sploiselle) [inline test/sqllogictest/map.slt:88]: sils vous plait, can I get tests for...
- _NULL_ keys both with _NULL_ present and absent from the map (idk how _NULL_ is handled as map elements,  so pardon my ignorance if that's not possible)
- looking for keys on empty map
- _NULL_ on an empty map
- the empty string as a variant on all of the same tests as _NULL_

File: src/sql/src/plan/func.rs [hunk 2]
```diff
@@ -520,6 +529,9 @@ pub enum ParamType {
     /// except it does not permit either `ScalarType::Array` or
     /// `ScalarType::List`.
     NonVecAny,
+    /// A polymorphic pseudotype permitting a `ScalarType::Map` of any element
```
Comment (sploiselle) [inline src/sql/src/plan/func.rs:532]: any value type?

File: src/sql/src/plan/func.rs [hunk 3]
```diff
@@ -1847,7 +1860,8 @@ lazy_static! {
                 })
             },
             "?" => Scalar {
-                params!(Jsonb, String) => JsonbContainsString
+                params!(Jsonb, String) => JsonbContainsString,
+                params!(NonVecAny, String) => JsonbContainsString
```
[the remark this answers is not in the document]
Comment (JLDLaughlin) [reply src/sql/src/plan/func.rs:1864]: ah yes!! fixing.

File: src/sql/src/plan/func.rs (hunk 2 above)
Comment (JLDLaughlin) [reply src/sql/src/plan/func.rs:532]: I've updated it to any non-nested value type!

File: test/sqllogictest/map.slt (hunk 1 above)
Comment (JLDLaughlin) [reply test/sqllogictest/map.slt:88]: NULL keys and empty maps do not parse as maps - I believe this covers the first three bullet points.

I have added tests for the empty string!

**Review by sploiselle — APPROVED** [review]
LGTM!

**Pull request merged**